### DISCUSSION TITLE
Best approach search URL route with changing parameters?

### DISCUSSION THREAD

**Discussion by MvdO79** [wheels-dev/wheels#1275] [Q&A] [answered]
What's the best approach for handling a search URL route that has changing parameters?

I have this URL:

 `http://127.0.0.1:60000/listings/9b1c6/marco/real+estate/apartments/flats/nederland/zh/alphen+aan+den+rijn/vve+balkon`

All parameters after "http://127.0.0.1:60000/listings/9b1c6/" are optional and changing.

The URL is pointing to a search function for which the parameters are used.

So how to setup a route for this?

**Comment by edthetechie** [comment]
Which version of CFWheels are you using? If it's 2.4 then have you tried
creating a really long route for the URL? I've made some long ones before
but never anything that long so I can't say if it will work or not. If that
doesn't work you could also try some kind of webserver URL rewrite rule to
catch the URL and deal with it that way.

**Reply by dbelanger** [reply, marked as answer]
I believe you are looking for the [wildcard segments](https://guides.cfwheels.org/cfwheels-guides/handling-requests-with-controllers/routing#wildcard-segments).  It would look something like this: 

`.get(name="listings", pattern="listings/[key]/*[searchOptions]", to="listings##list")`

Now, this wildcard segment `*[searchOptions]` will return **PARAMS.searchOptions** to your controller and you'll have to treat it like a "/" delimited list or, even better than a list, use the `listToArray( PARAMS.searchOptioins , '/' )` to convert that list to an array to use for your search.

**Reply by MvdO79** [reply]
Great!

**Comment by edthetechie** [comment]
Thanks David, I learned something new today as well :-)
